source/cellular_bg96_api.c

```diff
@@ -3204,3 +3381,1175 @@ CellularError_t Cellular_Init( CellularHandle_t * pCellularHandle,
+
+        if( optionLevel == CELLULAR_SOCKET_OPTION_LEVEL_SECURE )
+        {
+            if( option == CELLULAR_SOCKET_OPTION_SSL_CONTEXT_ID )
+            {
+                if( ( pSocketContext->socketState == SOCKETSTATE_ALLOCATED ) && ( optionValueLength == sizeof( uint8_t ) ) )
+                {
+                    pSocketSslConfig->sslContextId = *pOptionValue;
+                    cellularStatus = CELLULAR_SUCCESS;
+                }
+                else
+                {
+                    LogError( ( "Cellular_SocketSetSockOpt: Cannot change the sslContextID in this state %d or length %d is invalid.",
+                                pSocketContext->socketState, optionValueLength ) );
+                    cellularStatus = CELLULAR_INTERNAL_FAILURE;
+                }
+            }
+            else if( option == CELLULAR_SOCKET_OPTION_SSL_USAGE )
+            {
+                if( ( pSocketContext->socketState == SOCKETSTATE_ALLOCATED ) && ( optionValueLength == sizeof( uint8_t ) ) )
+                {
+                    pSocketSslConfig->useSsl = *pOptionValue;
+                    cellularStatus = CELLULAR_SUCCESS;
+                }
+                else
+                {
+                    LogError( ( "Cellular_SocketSetSockOpt: Cannot change the useSsl in this state %d or length %d is invalid.",
+                                pSocketContext->socketState, optionValueLength ) );
+                    cellularStatus = CELLULAR_INTERNAL_FAILURE;
+                }
+            }
+            else
+            {
+                /* This option is also not supported in port. */
+            }
+        }
+        else
+        {
+            /* This option level is also not supported in port. */
+        }
+    }
+    return cellularStatus;
+}
+
+CellularError_t Cellular_UploadFileToStorage( CellularHandle_t cellularHandle,
+                                              const char* filename,
+                                              const char* fileContent,
+                                              uint32_t fileSize,
+                                              uint32_t* pSentDataLength )
+{
+    CellularContext_t * pContext = ( CellularContext_t * ) cellularHandle;
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+    char atCommandBuffer[2*CELLULAR_AT_CMD_TYPICAL_MAX_SIZE] = {0};
+
+    CellularAtReq_t atReqUploadFile =
+    {
+        atCommandBuffer,
+        CELLULAR_AT_NO_RESULT,
+        NULL,
+        NULL,
+        NULL,
+        0,
+    };
+
+    CellularAtDataReq_t atDataReqUploadFile =
+    {
+        (const uint8_t *)fileContent,
+        fileSize,
+        pSentDataLength,
+        NULL,
+        0
+    };
+
+    if( ( cellularHandle == NULL ) || ( filename == NULL ) || ( fileContent == NULL ) || ( pSentDataLength == NULL ) )
+    {
+        return CELLULAR_BAD_PARAMETER;

```

Comment:
If you set **cellularStatus** here to **CELLULAR_BAD_PARAMETER** and put line 3507 in the else then you can do away with this return and only have a single return at the end of the function. One of the other things coming from MISRA as well. This is to avoid debugging a large function and never getting to your breakpoint, because there happens to be a return statement above it.